src/main/java/roomescape/ReservationController.java

```diff
@@ -0,0 +1,65 @@
+package roomescape;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+@Controller
+public class ReservationController {
+
+    private List<Reservation> reservations = new ArrayList<>();
+    private AtomicLong index = new AtomicLong(1);
+
+    @RequestMapping("/reservation")
+    public String reservation() {
+        return "reservation";
+    }
+
+    // 예약 목록 조회
+    @GetMapping("/reservations")
+    public ResponseEntity<List<Reservation>> read() {
+        return ResponseEntity.ok().body(reservations);
+    }
+
+    // id에 따른 예약 조회
+    @GetMapping("/reservations/{id}")
+    public ResponseEntity<Reservation> readEach(@PathVariable Long id) {
+        Reservation reservation = reservations.stream()
+                .filter(it -> Objects.equals(it.getId(), id))
+                .findFirst()
+                .orElseThrow(NoSuchElementException::new);
+        return ResponseEntity.ok().body(reservation);
+    }
+
+    // 예약 생성
+    @PostMapping("/reservations")
+    public ResponseEntity<Void> create(@RequestBody Reservation reservation) {
+        if (reservation.getName().isEmpty() || reservation.getTime().isEmpty() || reservation.getDate().equals("")){
+            throw new IllegalArgumentException("필요한 인자가 없습니다");
+        }
+        Reservation newReservation = Reservation.toEntity(reservation, index.getAndIncrement());
+        reservations.add(newReservation);
+        return ResponseEntity.created(URI.create("/reservations/" + newReservation.getId())).build();
```

Comment:
`Post` 요청을 했을 경우, 생성된 `Reservation`을 반환해야 JSON 타입으로 반환이 될 것 같아요!
현재 코드는 201 상태코드와 응답 헤더에 `Location : /reservations/{id}` 만 담긴 상태로 반환이 되고 있어요.
다음과 같이 응답 바디에 생성된 `Reservation` 객체를 넣어주면 JSON 타입으로 응답이 가게 될 거고, 3단계 테스트 코드도 주석을 제거했을 때 통과할 수 있습니다!
```suggestion
        return ResponseEntity
                .created(URI.create("/reservations/" + newReservation.getId()))
                .body(newReservation);
```